variables.tf

```diff
@@ -0,0 +1,80 @@
+/*
+Module variables
+*/
+
+variable "alb_is_internal" {
+  description = "Determines if the ALB is internal. Default: false"
+  default     = false
+}
+
+variable "alb_name" {
+  description = "The name of the ALB as will show in the AWS EC2 ELB console."
+  default     = "my-alb"
+}
+
+variable "alb_protocols" {
+  description = "A comma delimited list of the protocols the ALB accepts. e.g.: HTTPS"
+  default     = "HTTPS"
+}
+
+variable "alb_security_groups" {
+  description = "A comma separated string of security groups with which we associate the ALB. e.g. 'sg-edcd9784,sg-edcd9785'"
+  type        = "list"
+}
+
+variable "aws_region" {
+  description = "AWS region to use."
+}
+
+variable "backend_port" {
+  description = "The port the service on the EC2 instances listen on."
+  default     = 80
+}
+
+variable "backend_protocol" {
+  description = "The protocol the backend service speaks. Options: HTTP, HTTPS, TCP, SSL (secure tcp)."
+  default     = "HTTP"
```

Comment:
Same here, don't use default.

Reply:
I don't completely follow the logic of not providing any defaults... sane and often used defaults are a good thing to offer as they are in part what adds value to modules. If literally every variable is stripped of defaults, the result is a module that is pretty well as long as specifying all the components by hand without a module.

Can you make a better case for not providing defaults like this?